node/core/candidate-validation/src/lib.rs

```diff
@@ -657,6 +654,27 @@ impl ValidationBackend for ValidationHost {
 	}
 }
 
+// Tries executing a PVF a single time (no retries).
+async fn execute_pvf_once(
```

Comment:
Doesn't this fn look like a good candidate for inclusion into `ValidationHost` implementation (not into `ValidationBackend` trait though)?

Reply:
Why `ValidationHost`? I think I agree with Chris that it should be in `ValidationBackend` as `ValidationHost` already has `execute_pvf`. However, we need to do some extra stuff around calling `execute_pvf`.